compiler/test/compilable/test23923.c

```diff
@@ -0,0 +1,14 @@
+// https://issues.dlang.org/show_bug.cgi?id=23346
+
+struct S1 {
+    unsigned d:31;
+    int e:1;
+};
+
+struct S2 {
+    unsigned d:31;
+    _Bool e:1;
+};
```

Comment:
Missing test.
```suggestion
struct S2 {
    unsigned d:31;
    _Bool e:1;
};

struct S3 {
    unsigned d:31;
    char e:1;
};
```

Reply:
Interestingly, dmd and VC produce 4,8,8 for the sizes.

Reply:
added tests

Reply:
> Interestingly, dmd and VC produce 4,8,8 for the sizes.

That's because the rules of Microsoft Visual C/C++ are a bit-field won’t share the same storage unit with the previous bit-field if their underlying types have different sizes, and the bit-field will be aligned to the highest alignment of the underlying types of itself and of the previous bit-field.